Socket.php

```diff
@@ -100,9 +280,44 @@ public static function transportFactory($socketUri)
                     ? 994
                     : 6667);
 
+        $parsed = array_merge([
+            'path'     => '',
+            'fragment' => '',
+            'query'    => '',
+            'user'     => null,
+            'pass'     => null
+        ], $parsed);
+        $complement =
+            ltrim($parsed['path'], '/') .
+            $parsed['fragment'] .
+            (empty($parsed['query'])?'':'?' . $parsed['query']);
+
+        $entity   = null;
+        $flags    = null;
+        $options  = [];
+
+        $pattern  = '/^((#|%23)?([^,?]+))(,([^?]+))?(\?(.*))?$/';
```

Comment:
In the RFC, it's mentioned that a client must match an encoded # if present... Maybe I understand it wrong : https://tools.ietf.org/html/draft-butcher-irc-url-04#section-2.5.1